internal/domain/domain.go

```diff
@@ -23,7 +24,7 @@ func AppDefaultRoute(ctx context.Context, name string) (string, error) {
 	if err != nil {
 		return "", err
 	}
-	return fmt.Sprintf("%s.%s", name, mainDomain), nil
+	return fmt.Sprintf("%s-%s.%s", name, names.RandomDNSString(5), mainDomain), nil
```

Comment:
Would it make sense to compute a truncated hash from the namespace here instead of full random ?

Reply:
Hum, it could be another solution. The only "problem" would be that with a collision then you will have to provide a custom route to unblock the deployment.

i.e.: the hash of the `abc123` and the `xxx555` namespaces is for both `dnm8i34j9o0edji890we`, and the suffix will end to be the same. But to have a conflict the app name needs to be the same, and the probability to have this collision is VERYVERYVERY low. So it's fine for me to add this solution instead of the random one.

At least we avoid to import other libraries, and have a deterministic solution. :+1: